### ISSUE TITLE
"[name].js?[hash]" file name not handled correctly in v2

### ISSUE THREAD

**Issue by Daniel15** [ztoben/assets-webpack-plugin#19]
My Webpack config has something like this in it:

```
  entry: {
    desktop: './js/desktop.js',
    mobile: './js/mobile.js'
  },
  output: {
    path: path.join(__dirname, 'public', 'assets', 'cache'),
    filename: process.env.NODE_ENV === 'production'
      ? '[name].min.js?[hash]'
      : '[name].js?[hash]',
  },
```

Output from 0.3.1:

```
{"desktop":"desktop.js?177b0e49e38f148eb91a","mobile":"mobile.js?177b0e49e38f148eb91a"}
```

This is easy, I just read the `desktop` property

Output from 2.2.0:

```
{"desktop":{"js?177b0e49e38f148eb91a":"desktop.js?177b0e49e38f148eb91a","jsMap":"desktop.js.map?177b0e49e38f148eb91a"},"mobile":{"js?177b0e49e38f148eb91a":"mobile.js?177b0e49e38f148eb91a","jsMap":"mobile.js.map?177b0e49e38f148eb91a"}}
```

What do I do here? The key has the hash in it :/ I could do the equivalent of `Object.keys(assets.desktop)[0]` but that's a bit ugly.

I've reverted to 0.3.1 in the meantime.

**Comment by sporto** [comment]
@Daniel15 oh, that looks really wrong, the hash is not meant to be in the key. Will look into this.

**Comment by Daniel15** [comment]
It looks like the `getFileExt` function is only removing querystrings from sourcemaps, perhaps it should be removing them from all files: https://github.com/sporto/assets-webpack-plugin/blob/master/index.js#L123-L125

**Comment by sporto** [comment]
@Daniel15 I published a new version 2.2.2, not sure if this fixes your issue, could you test it?

**Comment by Daniel15** [comment]
Yeah this looks good now, thanks! Sorry for the delay in replying.
